src/cswinrt/strings/WinRT.cs

```diff
@@ -448,31 +491,65 @@ public static unsafe (ObjectReference<IActivationFactoryVftbl> obj, int hr) GetA
         {
             Marshal.ThrowExceptionForHR(Platform.CoDecrementMTAUsage(_mtaCookie));
         }
-    }
-
+    }
+
     internal class BaseActivationFactory
     {
-        private readonly ObjectReference<IActivationFactoryVftbl> _IActivationFactory;
+        private volatile Tuple<ObjectReference<IActivationFactoryVftbl>, IntPtr> _IActivationFactory;
+
+        public ObjectReference<IActivationFactoryVftbl> Value
+        {
+            get
+            {
+                var existingInstance = _IActivationFactory;
+                if (existingInstance != null && (existingInstance.Item2 == IntPtr.Zero || existingInstance.Item2 == Context.GetContextToken()))
+                {
+                    return existingInstance.Item1;
+                }
+
+                var newFactory = InitializeFactory();
+                _IActivationFactory = newFactory;

```

Comment:
Q: two racing threads might end up returning different instances of the activation factory. Is that a concern?

Reply:
That is not a concern. Caching it is opportunistic.